setup.py

```diff
@@ -0,0 +1,1473 @@
+    )
+    write_cuda_env_config_py(
+        filename=paddle_binary_dir + '/python/paddle/cuda_env.py'
+    )
+    write_distributed_training_mode_py(
+        filename=paddle_binary_dir
+        + '/python/paddle/fluid/incubate/fleet/parameter_server/version.py'
+    )
+
+    (
+        setup_requires,
+        packages,
+        scripts,
+        package_data,
+        package_dir,
+        ext_modules,
+        headers,
+    ) = get_setup_parameters()
+
+    # Log for PYPI, get long_description of setup()
+    with open(
+        paddle_source_dir + '/python/paddle/README.rst', "r", encoding='UTF-8'
+    ) as f:
+        long_description = f.read()
+
+    # strip *.so to reduce package size
+    if envir_var.get("WITH_STRIP") == 'ON':
+        command = (
+            'find '
+            + paddle_binary_dir
+            + '/python/paddle -name "*.so" | xargs -i strip {}'
+        )
+        if os.system(command) != 0:
+            raise Exception("strip *.so failed, command: %s" % command)
+
+    setup(
+        name=package_name,
+        version=paddle_version,
+        description='Parallel Distributed Deep Learning',
+        long_description=long_description,
+        long_description_content_type="text/markdown",
+        author_email="Paddle-better@baidu.com",
+        maintainer="PaddlePaddle",
+        maintainer_email="Paddle-better@baidu.com",
+        url='https://www.paddlepaddle.org.cn/',
+        download_url='https://github.com/paddlepaddle/paddle',
+        license='Apache Software License',
+        packages=packages,
+        install_requires=setup_requires,
+        ext_modules=ext_modules,
+        package_data=package_data,
+        package_dir=package_dir,
+        scripts=scripts,
+        distclass=BinaryDistribution,
+        headers=headers,
+        cmdclass={
+            'install_headers': InstallHeaders,
+            'install': InstallCommand,
+            'egg_info': EggInfo,
+            'install_lib': InstallLib,
+        },
+        entry_points={
+            'console_scripts': [
+                'fleetrun = paddle.distributed.launch.main:launch'
+            ]
+        },
+        classifiers=[
+            'Development Status :: 5 - Production/Stable',
+            'Operating System :: OS Independent',
+            'Intended Audience :: Developers',
+            'Intended Audience :: Education',
+            'Intended Audience :: Science/Research',
+            'License :: OSI Approved :: Apache Software License',
+            'Programming Language :: C++',
+            'Programming Language :: Python :: 2.7',
+            'Programming Language :: Python :: 3.5',
+            'Programming Language :: Python :: 3.6',
+            'Programming Language :: Python :: 3.7',
+            'Programming Language :: Python :: 3.8',
+        ],
+    )
+    if os.path.exists('${SETUP_LOG_FILE}'):
+        os.system('grep -v "purelib" ${SETUP_LOG_FILE}')
```

Comment:
why do it?

Reply:
This is the logic from the original setup.py.in file,  which can be removed